lineage/tHMM.py

```diff
@@ -193,3 +193,52 @@ def log_E_score(EL_array, state_tree_sequence):
     for idx, row in enumerate(log_EL_array):
         log_E_score_holder += row[state_tree_sequence[idx]]
     return log_E_score_holder
+
+
+def fit_list(tHMMobj_list, tolerance=1e-9, max_iter=1000):
+    """Runs the tHMM function through Baum Welch fitting for a list containing a set of data for different concentrations"""
+
+    # Step 0: initialize with KMeans and do an M step
+    # when there are no fixed emissions, we need to randomize the start
+    init_gammas = [[sp.multinomial.rvs(n=1, p=[1. / tHMMobj_list[0].num_states] * tHMMobj_list[0].num_states, size=len(lineage))
+                    for lineage in tHMMobj.X] for tHMMobj in tHMMobj_list]
+
+    MSD_list = []
+    betas_list = []
+    gammas_list = []
+    NF_list = []
+    new_LL = []
+
+    for _, tHMM in enumerate(tHMMobj_list):
+        init_gammas = [sp.multinomial.rvs(n=1, p=[1. / tHMM.num_states] * tHMM.num_states, size=len(lineage))
+                    for lineage in tHMM.X]
+
+        do_M_E_step(tHMM, init_gammas)
+
+        # Step 1: first E step
+        MSD, NF, betas, gammas = do_E_step(tHMM)
+        MSD_list.append(MSD)
+        NF_list.append(NF)
+        betas_list.append(betas)
+        gammas_list.append(gammas)
+
+        new_LL.append(np.sum(calculate_log_likelihood(NF)))
+
+    # sum over all populations in the list
+    total_new_LL = np.sum(new_LL)
+
+    # first stopping condition check
+    for _ in range(max_iter):
+        old_LL = total_new_LL
+
+        do_M_step_list(tHMMobj_list, MSD_list, betas_list, gammas_list)
+        for tHMM in tHMMobj_list:
+            MSD, NF, betas, gammas = do_E_step(tHMM)
```

Comment:
Doesn't this need to overwrite `MSD_list`, etc?

Reply:
Corrected.